src/RoleManager.sol

```diff
@@ -429,18 +440,16 @@ contract RoleManager is Positions {
      */
     function removeVault(
         address _vault
-    ) external virtual onlyPositionHolder(CZAR) {
-        // Get the vault specific config.
-        VaultConfig memory config = vaultConfig[_vault];
-        // Make sure the vault has been added to the role manager.
-        require(config.asset != address(0), "vault not added");
-
+    ) external virtual vaultIsAdded(_vault) onlyPositionHolder(CZAR) {
         // Transfer the role manager position.
         IVault(_vault).transfer_role_manager(chad);
 
         // Address of the vault to replace it with.
         address vaultToMove = vaults[vaults.length - 1];
 
+        // Get the vault specific config.
+        VaultConfig memory config = vaultConfig[_vault];
```

Comment:
You are only using the vault.config, you can only read the config and not copy the entire structure to memory.

Reply:
Nermind